Changed all Regexes to disallow spaces

RedingJeff opened the pull request:
Now the fields are really FIWARE safe. 

In two cases I shortened the regex, but as far as I saw (and the text stateted) these limitations were not needed for these fields. * is an allowed char.

tstorek reviewed (changes requested):
@RedingJeff 
Thanks for the pull request. Maybe this would be also a great opportunity to introduce a test for this to the changed models. Don't you think?

I think that * is part of the regex expression. Please, double check on e.g. https://www.regexpal.com/

RedingJeff commented:
I added tests for all the fields. 
To reduce the redundancy and improve maintainability, I grouped the two FiwareRegexes in a class togehter

I checked the simple regex again, the "*" is not needed, but the regex that should prevent specific strings was not correct. I corrected it.

tstorek commented on filip/models/base.py:
```diff
@@ -88,3 +88,13 @@ class FiwareHeader(BaseModel):
     class Config(BaseConfig):
         allow_population_by_field_name = True
         validate_assignment = True
+
```
Almost there. Maybe we can use annotated enum here. Hence, we would alogn it with the other enums used in the library.

tstorek reviewed (changes requested):
Hi. Looks good. Please raise the version number by a minor release and write a small changelog comment as Bugfix. Easiest way is to search the project files for the current version number and and then adjust it :)
